bundles/sirix-core/src/main/java/org/sirix/io/cloud/amazon/AmazonS3StorageReader.java

```diff
@@ -91,7 +91,9 @@ protected Path readObjectDataFromS3(String keyName) {
 
             ResponseBytes<GetObjectResponse> objectBytes = s3Client.getObjectAsBytes(objectRequest);
             byte[] data = objectBytes.asByteArray();
-            String path = System.getProperty("java.io.tmpdir")	+ FileSystems.getDefault().getSeparator() + keyName;
+            /*As the bucketName has to be same as the database name, it makes sense to use/create file on the local filesystem
+             * instead of in the tmp partition*/
+            String path = FileSystems.getDefault().getSeparator() + bucketName + FileSystems.getDefault().getSeparator() + keyName;
```

Comment:
We should use the path as defined in the storage class (databases are stored in user home in the sirix-data folder IIRC per default).